src/plugins/guided_onboarding/public/components/guide_button.tsx

```diff
@@ -39,12 +41,45 @@ const getStepNumber = (state: GuideState): number | undefined => {
 };
 
 export const GuideButton = ({
-  guideState,
+  pluginState,
   toggleGuidePanel,
   isGuidePanelOpen,
+  navigateToLandingPage,
 }: GuideButtonProps) => {
-  const stepNumber = getStepNumber(guideState);
-  const stepReadyToComplete = guideState.steps.find((step) => step.status === 'ready_to_complete');
+  // TODO handle loading, error state
+  // https://github.com/elastic/kibana/issues/139799, https://github.com/elastic/kibana/issues/139798
+
+  // if there is no active guide
+  if (!pluginState || !pluginState.activeGuide || !pluginState.activeGuide.isActive) {
```

Comment:
Do we need the `isActive` flag anymore since the plugin state provides the `activeGuide`? In other words, would the `activeGuide` ever not be active? 🤔

Reply:
I thought about that too, but I think we still need the flag `isActive` because guide are a separate saved object type and we keep all guides with their states in there. The only way to know which is the active guide is by querying the flag. Alternatively, we could keep an ID of the active guide in the plugin state SO, but I think the query works pretty well and is an already "tested" approach.